app/screens/people-screen/circles/share-circles-image.tsx

```diff
@@ -0,0 +1,225 @@
+import { useCirclesQuery, WelcomeProfile } from "@app/graphql/generated"
+import { makeStyles, Text, useTheme } from "@rneui/themed"
+import { forwardRef, useMemo, useRef } from "react"
+import { View, Share as NativeShare } from "react-native"
+import Icon from "react-native-vector-icons/Ionicons"
+
+import Logo from "@app/assets/logo/app-logo-dark.svg"
+
+import { captureRef } from "react-native-view-shot"
+import Share from "react-native-share"
+import { Circle } from "@app/components/circle"
+import { LinearGradient } from "react-native-linear-gradient"
+import crashlytics from "@react-native-firebase/crashlytics"
+import { useI18nContext } from "@app/i18n/i18n-react"
+import { getInviteLink } from "@app/config/appinfo"
+
+export const CirclesDashboardHeaderRight: React.FC = () => {
+  const shareImgRef = useRef<View | null>(null)
+  const styles = useStyles()
+  const { LL } = useI18nContext()
+
+  const { data } = useCirclesQuery()
+
+  const username = data?.me?.username || ""
+  const welcomeProfile = data?.me?.defaultAccount.welcomeProfile
+
+  const ShareImg = useMemo(() => {
+    if (welcomeProfile)
+      return (
+        <ShareImageComponent
+          ref={shareImgRef}
+          username={username}
+          welcomeProfile={welcomeProfile}
+        />
+      )
+    return <></>
+  }, [username, welcomeProfile])
+
+  const share = async () => {
+    try {
+      if (welcomeProfile?.innerCircleAllTimeCount === 0) {
+        const inviteLink = getInviteLink(data?.me?.username || "")
```

Comment:
instead of passing an empty string to `getInviteLink`  with `data?.me?.username || ""`, I would make `getInviteLink` to accept undefined and manage the edge case there. (just a comment... don't have to do it now)